### DISCUSSION TITLE
Laravel Livewire Table not refreshing with $this->refresh;

### DISCUSSION THREAD

**Discussion by dah-HUB** [rappasoft/laravel-livewire-tables#996] [Q&A]
Hi there,

I have a table that displays data bfiltered by date (default '') ) using the builder function,

public function builder(): Builder {
    ....
    ->when(!empty($this->from), function ($q) { return $q->where('start', '>=', $this->from); })
    ....

the date is sent from a  js component :
    Livewire.emit('DateRangeSelected', date1.format('YYYY-MM-DD'))

The listener in the table is like :
    public function DateRangeSelected($from = ''){
        $this->from = $from;
        $this->refresh;

All is working well, except the table is not refreshing, 
I have to click in any table header so that the data is reordered and rendered,

note: The table was on a tailwind tab, so i put it out of it, but the same, the table is not refreshing

plz advice
Thanks

**Comment by lrljoe** [comment]
Have you tried
$this->emit('refreshDatatable')

This is the default listener to refresh the data table.

**Reply by dah-HUB** [reply]
Hi @lrljoe 

Thx for the replay,
Yes that was the first thing I did,
Btw, refreshDatatable is a listener for refresh :

    protected $listeners = [
        'refreshDatatable' => '$refresh',
        'setSort' => 'setSortEvent',
        'clearSorts' => 'clearSortEvent',
        'setFilter' => 'setFilterEvent',
        'clearFilters' => 'clearFilterEvent',
    ];
:+1:

**Reply by dah-HUB** [reply]
(After some reading in the /vendor/)
To refresh from inside the Table itself , I set : $this->refresh = true;
Don't think it's the best way to do it, but its working :)

@lrljoe
I was calling $this->emit('refreshDatatable') from the table, so that it was not working and its logical :(
Thx

**Reply by lrljoe** [reply]
Call $emitSelf instead, to a set listener (rather than just refresh) and it'll work
